WISE-Community/WISE
# Allow admin to edit site announcement using admin tools

Admins can now set the site announcement banner and content via `/admin/portal/manage`. I also updated the legacy unit warning messages to reflect pending deprecation.

To test:
- Add 'announcement' column to 'portal' table in local db and add default data:
```
ALTER TABLE portal
ADD announcement MEDIUMTEXT;
UPDATE portal SET announcement = '{"visible": false,"bannerText": "","bannerButton": "","title": "","content": "","buttons": []}' WHERE id = 1;
```
- Load site homepage and make sure no announcement banner appears.
- Log in as an admin and navigate to `/admin/portal/manage`.
- Edit the announcement data and save. For example:
```
{
	"visible": true,
	"bannerText": "Legacy unit support is ending.",
	"bannerButton": "Learn More",
	"title": "Ending support for legacy units",
	"content": "<p>On October 1, 2019, we will be removing support for legacy WISE units. While WISE users will still be able to view legacy unit content in preview mode and review student work from previous classroom implementations, they will no longer be able to create new legacy units or run legacy units with their students.</p>",
	"buttons": [{
		"text": "Contact Us",
		"link": "https://wise.berkeley.edu/contact",
                "color": "accent"
	}]
}
```
- Navigate to site home and make sure the announcement shows up and that clicking the "Learn More" button opens the announcement details dialog.
- Create a new run using a legacy (wise4) unit.
- Make sure the text "(Legacy Unit)" appears next to the new run's title in the teacher "Class Schedule".

Resolves #2010.
